org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporterTest.java

```diff
@@ -95,4 +101,70 @@ public void testGradlePersistence() throws Exception {
 		assertTrue(GradleProjectImporter.shouldSynchronize(project.getLocation().toFile()));
 	}
 
+	@Test
+	public void testWorkspaceSettings() throws Exception {
+		File file = null;
+		Map<String, String> env = System.getenv();
+		Map<String, String> oldEnv = new HashMap<>();
+		oldEnv.putAll(env);
+		String gradleHomeProp = System.getProperty(GradleProjectImporter.GRADLE_HOME);
+		try {
+			System.clearProperty(GradleProjectImporter.GRADLE_HOME);
+			Map<String, String> newEnv = new HashMap<>();
+			newEnv.putAll(oldEnv);
+			newEnv.remove(GradleProjectImporter.GRADLE_HOME);
+			setEnv(newEnv);
+			IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
+			File rootFile = root.getLocation().toFile();
+			file = new File(rootFile, "fakeGradleHome");
+			System.setProperty(GradleProjectImporter.GRADLE_HOME, file.getAbsolutePath());
+			boolean overrideWorkspaceSettings = GradleProjectImporter.getGradleHomeFile() != null;
+			assertFalse(overrideWorkspaceSettings);
+			file.mkdir();
+			file.deleteOnExit();
+			overrideWorkspaceSettings = GradleProjectImporter.getGradleHomeFile() != null;
+			assertTrue(overrideWorkspaceSettings);
+		} finally {
+			if (gradleHomeProp == null) {
+				System.clearProperty(GradleProjectImporter.GRADLE_HOME);
+			} else {
+				System.setProperty(GradleProjectImporter.GRADLE_HOME, gradleHomeProp);
+			}
+			setEnv(oldEnv);
+			if (file != null) {
+				file.delete();
+			}
+		}
+	}
+
+	protected static void setEnv(Map<String, String> newEnv) throws Exception {
```

Comment:
This is way to complex and brittle for what we want to test.

I suggest you refactor getGradleHomeFile() to make it more testable, like getGradleHomeFile(Map env, Properties sysprops)